recipes/7zip/19.00/conanfile.py

```diff
@@ -55,47 +77,40 @@ def _msvc_platform(self):
         }[str(self.settings.arch)]
 
     def _build_msvc(self):
-        with tools.vcvars(self.settings):
-            with chdir(self, os.path.join("CPP", "7zip")):
-                self.run(f"nmake /f makefile PLATFORM={self._msvc_platform}")
+        self.run(f"nmake /f makefile PLATFORM={self._msvc_platform}", cwd=os.path.join(self.source_folder, "CPP", "7zip"))
 
     def _build_autotools(self):
         # TODO: Enable non-Windows methods in configure
-        autotools = AutoToolsBuildEnvironment(self)
-        extra_env = {}
-        if self.settings.os == "Windows" and self.settings.compiler == "gcc":
-            extra_env["IS_MINGW"] = "1"
-        with tools.environment_append(extra_env):
-            with chdir(self, os.path.join("CPP", "7zip", "Bundles", "LzmaCon")):
-                autotools.make(args=["-f", "makefile.gcc"], target="all")
+        autotools = Autotools(self)
+        with chdir(self, os.path.join(self.source_folder, "CPP", "7zip", "Bundles", "LzmaCon")):
+            autotools.make(args=["-f", "makefile.gcc"], target="all")
 
     def _patch_sources(self):
-        if self.settings.compiler == "Visual Studio":
-            fn = os.path.join("CPP", "Build.mak")
+        if is_msvc(self):
+            fn = os.path.join(self.source_folder, "CPP", "Build.mak")
             os.chmod(fn, 0o644)
             replace_in_file(self, fn, "-MT", f"-{self.settings.compiler.runtime}")
             replace_in_file(self, fn, "-MD", f"-{self.settings.compiler.runtime}")
 
     def build(self):
         self._patch_sources()
-        if self.settings.compiler == "Visual Studio":
+        if is_msvc(self):
             self._build_msvc()
         else:
             self._build_autotools()
 
     def package(self):
-        self.copy("DOC/License.txt", src="", dst="licenses")
-        self.copy("DOC/unRarLicense.txt", src="", dst="licenses")
+        copy(self, "License.txt", dst=os.path.join(self.package_folder, "licenses"), src=os.path.join(self.source_folder, "DOC"))
+        copy(self, "unRarLicense.txt", dst=os.path.join(self.package_folder, "licenses"), src=os.path.join(self.source_folder, "DOC"))
```

Comment:
Does this license only apply to the DLL? If we don't package the DLL, then maybe we don't need it?

Reply:
No, it is explicitly mentioned in the `License.txt` and also relates to the source code:
```
  The licenses for files are:

    1) CPP/7zip/Compress/Rar* files: the "GNU LGPL" with "unRAR license restriction"
    2) CPP/7zip/Compress/LzfseDecoder.cpp: the "BSD 3-clause License"
    3) Some files are "public domain" files, if "public domain" status is stated in source file.
    4) the "GNU LGPL" for all other files. If there is no license information in 
       some source file, that file is under the "GNU LGPL".

  The "GNU LGPL" with "unRAR license restriction" means that you must follow both 
  "GNU LGPL" rules and "unRAR license restriction" rules.
```

And the generated binary contains rar support:
```
7z i

...
 0  ...F..................  Rar      rar r00       R a r ! 1A 07 00
 0  ...F..................  Rar5     rar r00       R a r ! 1A 07 01 00
...
```

Reply:
Thanks, I wasn't sure since their license web page says things differently: https://www.7-zip.org/license.txt